gstd/gstd_unix.c

```diff
@@ -0,0 +1,298 @@
+    guint property_id, GValue * value, GParamSpec * pspec)
+{
+  GstdUnix *self = GSTD_UNIX (object);
+
+  switch (property_id) {
+    case PROP_UNIX_PATH:
+      GST_DEBUG_OBJECT (self, "Returning unix-path %s", self->unix_path);
+      g_value_set_string (value, self->unix_path);
+      break;
+
+    case PROP_UNIX_NUM_PORTS:
+      GST_DEBUG_OBJECT (self, "Returning number-ports %u", self->num_ports);
+      g_value_set_uint (value, self->num_ports);
+      break;
+
+	  default:
+      /* We don't have any other property... */
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+      break;
+  }
+}
+
+static void
+gstd_unix_set_property (GObject * object,
+    guint property_id, const GValue * value, GParamSpec * pspec)
+{
+  GstdUnix *self = GSTD_UNIX (object);
+
+  switch (property_id) {
+    case PROP_UNIX_PATH:
+      GST_DEBUG_OBJECT (self, "Changing unix-path current value: %s",
+          self->unix_path);
+      gstd_unix_set_path(self, g_value_get_string (value));
+      break;
+
+    case PROP_UNIX_NUM_PORTS:
+      GST_DEBUG_OBJECT (self, "Changing num-ports current value: %u",
+          self->num_ports);
+      self->num_ports = g_value_get_uint (value);
+      GST_DEBUG_OBJECT (self, "Value changed %u", self->num_ports);
+      break;
+
+    default:
+      /* We don't have any other property... */
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, property_id, pspec);
+      break;
+  }
+}
+
+
+static void
+gstd_unix_dispose (GObject * object)
+{
+  GstdUnix *self = GSTD_UNIX (object);
+  GstdIpc *parent = GSTD_IPC (object);
+  guint i;
+
+  GST_INFO_OBJECT (object, "Deinitializing gstd UNIX");
+
+  if (parent->enabled) {
+    for (i = 0; i < self->num_ports; i++) {
+      gchar * path_name = g_strdup_printf ("%s_%d", self->unix_path, i);
+
+      if (unlink(path_name) != 0) {
+        GST_ERROR_OBJECT (object, "Unable to delete UNIX path (%s)", strerror(errno));
+      }
+      g_free (path_name);
+    }
+  }
+
+  G_OBJECT_CLASS (gstd_unix_parent_class)->dispose (object);
+}
+
+GstdReturnCode
+gstd_unix_add_listener_address (GstdSocket * base, GSocketService ** service)
+{
+  guint debug_color;
+  GError *error = NULL;
+  GstdUnix *self = GSTD_UNIX (base);
+  gchar* path = self->unix_path;
+  guint i;
+
+  GST_DEBUG_OBJECT (self, "Getting UNIX Socket address");
+
+  if (!gstd_unix_debug) {
+    /* Initialize debug category with nice colors */
+    debug_color = GST_DEBUG_FG_BLACK | GST_DEBUG_BOLD | GST_DEBUG_BG_WHITE;
+    GST_DEBUG_CATEGORY_INIT (gstd_unix_debug, "gstdunix", debug_color,
+        "Gstd UNIX category");
+  }
+
+  *service = g_threaded_socket_service_new (self->num_ports);
+
+  for (i = 0; i < self->num_ports; i++) {
+    GSocketAddress *address;
+    gchar * path_name = g_strdup_printf ("%s_%d", path, i);
+
+    address = g_unix_socket_address_new (path_name);
+	  g_free (path_name);
+
+    g_socket_listener_add_address (G_SOCKET_LISTENER (*service),
```

Comment:
Please validate what happens if two different gstd instances try to open the same sockets.